syntax = "proto3";

package total;

/////////////////////////////////////////

service serviceguide {
  rpc Getservice(service_name) returns (service_add) {}
}
message service_add {
  string address = 1;
  string port = 2;
  string name= 3;
}
message service_name {
  string name = 1;
}

////////////////////////////////////////

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}

//////////////////////////////////////////

service Calculator {

	rpc Add (AddRequest) returns (AddReply) {}
	rpc Substract (SubstractRequest) returns (SubstractReply) {}
	rpc Multiply (MultiplyRequest) returns (MultiplyReply) {}
	rpc Divide (DivideRequest) returns (DivideReply) {}
}

message AddRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message AddReply{
	
	int32 n1=1;
}

message SubstractRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message SubstractReply{
	
	int32 n1=1;
}

message MultiplyRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message MultiplyReply{
	
	int32 n1=1;
}

message DivideRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message DivideReply{
	
	float f1=1;
}
	
//////////////////////////////////////////

message Text {
    string value = 1;
    string dest=2;
}
message returntext{
    string value=1;
}

service Translator {
    rpc GoogTrans(Text) returns (returntext) {}
    
}
    
//////////////////////////////////////////
